sgkit/window.py

```diff
@@ -0,0 +1,174 @@
+from typing import Any, Callable
+
+import dask.array as da
+import numpy as np
+from xarray import Dataset
+
+from sgkit.utils import conditional_merge_datasets
+
+# Window definition (user code)
+
+
+def window(
+    ds: Dataset,
+    size: int,
+    step: int,
+    merge: bool = True,
+) -> Dataset:
+    """Add windowing information to a dataset."""
+
+    n_variants = ds.dims["variants"]
+
+    length = n_variants
+    window_starts, window_stops = _get_windows(0, length, size, step)
+
+    new_ds = Dataset(
+        {
+            "window_start": (
+                "windows",
+                window_starts,
+            ),
+            "window_stop": (
+                "windows",
+                window_stops,
+            ),
+        }
+    )
+    return conditional_merge_datasets(ds, new_ds, merge)
+
+
+def _get_windows(start: int, stop: int, size: int, step: int) -> Any:
+    # Find the indexes for the start positions of all windows
+    # TODO: take contigs into account
```

Comment:
Worth elevating as an issue?  I'm sure it will come up again in the context of LD estimation.

Reply:
Opened #335